### DISCUSSION TITLE
我想要用opencompass测评Ziya_llama_13B模型，请教各位大佬，配置文件应该怎么修改？

### DISCUSSION THREAD

**Discussion by Elsablabla** [open-compass/opencompass#677] [Q&A]
调用ziya推理的代码如下：
`from transformers import AutoTokenizer
from transformers import LlamaForCausalLM
import torch

device = torch.device("cuda")
ckpt = '/workspace/pc/Ziya-LLaMA-13B-v1'

query="帮我写一份西安旅游计划"
model = LlamaForCausalLM.from_pretrained(ckpt, torch_dtype=torch.float16, device_map="auto")
tokenizer = AutoTokenizer.from_pretrained(ckpt, use_fast=False)
inputs = '<human>:' + query.strip() + '\n<bot>:'

input_ids = tokenizer(inputs, return_tensors="pt").input_ids.to(device)
generate_ids = model.generate(
            input_ids,
            max_new_tokens=1024,
            do_sample = True,
            top_p = 0.85,
            temperature = 1.0,
            repetition_penalty=1.,
            eos_token_id=2,
            bos_token_id=1,
            pad_token_id=0)
output = tokenizer.batch_decode(generate_ids)[0]
print(output)
`

**Comment by yingfhu** [comment]
```python
    models = [dict(
        type=HuggingFaceCausalLM,
        abbr='Ziya_llama_13B',
        path="/workspace/pc/Ziya-LLaMA-13B-v1",
        tokenizer_path='/workspace/pc/Ziya-LLaMA-13B-v1',
        tokenizer_kwargs=dict(padding_side='left',
                              truncation_side='left',
                              use_fast=False,
                              ),
        max_out_len=1024,
        max_seq_len=2048,
        batch_size=8,
        model_kwargs=dict(device_map='auto'),
        generation_kwargs=dict(
            repetition_penalty=1.,
            do_sample=True,
            top_p=0.85,
            temperature=1.0,
        )
        run_cfg=dict(num_gpus=2, num_procs=1),
    )]
```
something like this, you can adjust it if encounter any problem.

**Reply by Elsablabla** [reply]
wow! thanks for your help! Actually, i am confused about
`from transformers import AutoTokenizer
from transformers import LlamaForCausalLM`,
can `type=HuggingFaceCausalLM,` substitute the above imports?

**Reply by yingfhu** [reply]
use `from opencompass.models import HuggingFaceCausalLM` in config should be fine

**Reply by Elsablabla** [reply]
Great, I'm trying it! Thank you very much for your help. It was also yours help last time and helped me run through the model! I'm a novice, thank you very much for your patient and quick help!
